src/extension/app/store/app.js

```diff
@@ -355,12 +357,25 @@ export class AppStore {
             };
             return environments.some((env) => envChecks[env] && envChecks[env].call(appStore));
           };
-            // assemble plugin config
+          // assemble plugin config
           const plugin = {
             custom: true,
-            id,
+            id: id || `custom-plugin-${i}`,
```

Comment:
Should get a default value during assignment above

```suggestion
            id,
```